resources/js/common/components/GlobalSearch/components/SearchResults/SearchResults.tsx

```diff
@@ -0,0 +1,194 @@
+import type { FC } from 'react';
+import { useTranslation } from 'react-i18next';
+import { FaGamepad } from 'react-icons/fa';
+import { ImTrophy } from 'react-icons/im';
+import type { IconType } from 'react-icons/lib';
+import { LuNetwork, LuUsers } from 'react-icons/lu';
+
+import { BaseCommandGroup, BaseCommandItem } from '@/common/components/+vendor/BaseCommand';
+import type { useSearchQuery } from '@/common/hooks/queries/useSearchQuery';
+
+import type { SearchMode } from '../../models/search-mode.model';
+import { AchievementResultDisplay } from './AchievementResultDisplay';
+import { GameResultDisplay } from './GameResultDisplay';
+import { HubResultDisplay } from './HubResultDisplay';
+import { UserResultDisplay } from './UserResultDisplay';
+
+// TODO switch all window.location.assign to use router.visit() and strongly-typed routes
+// - this is not currently possible because Inertia/Ziggy are unavailable in Blade contexts
+
+type SearchResult =
+  | App.Data.User
+  | App.Platform.Data.Game
+  | App.Platform.Data.Achievement
+  | App.Platform.Data.GameSet;
+
+interface SearchSection {
+  key: string;
+  heading: string;
+  results: SearchResult[];
+  relevance: number;
+  limit: number;
+  icon: IconType;
+  render: (item: SearchResult) => React.ReactNode;
+}
+
+interface SearchResultsProps {
+  currentSearchMode: SearchMode;
+  searchResults: ReturnType<typeof useSearchQuery>['data'];
+  onClose: () => void;
+}
+
+export const SearchResults: FC<SearchResultsProps> = ({
+  currentSearchMode,
+  searchResults,
+  onClose,
+}) => {
+  const { t } = useTranslation();
+
+  if (!searchResults) {
+    return null;
+  }
+
+  const sections: SearchSection[] = [
+    {
+      key: 'users',
+      heading: t('Users'),
+      results: searchResults.results.users || [],
+      relevance: searchResults.scopeRelevance.users || 0,
+      limit: 3,
+      icon: LuUsers,
+
+      render: (user) => {
+        const safeUser = user as App.Data.User;
+
+        return (
+          <BaseCommandItem
+            key={`user-${safeUser.displayName}`}
+            onSelect={() => {
+              window.location.assign(`/user/${safeUser.displayName}`);
+              onClose();
+            }}
+          >
+            <UserResultDisplay user={safeUser} />
+          </BaseCommandItem>
+        );
+      },
+    },
+    {
+      key: 'games',
+      heading: t('Games'),
+      results: searchResults.results.games || [],
+      relevance: searchResults.scopeRelevance.games || 0,
+      limit: 6,
+      icon: FaGamepad,
+
+      render: (game) => (
+        <BaseCommandItem
+          key={`game-${game.id}`}
+          className="group"
+          onSelect={() => {
+            window.location.assign(`/game/${game.id}`);
+            onClose();
```

Comment:
There's no URL preview when hovering over a link. While most users probably don't care, it helps to explain why some records are being returned by ID when searching on numbers.
![image](https://github.com/user-attachments/assets/f7277bee-09cb-4bdb-8e0d-3d3afd17e3aa)
I search for "2025" a lot to get to the event pages. Here's the top 6 results I see:
![image](https://github.com/user-attachments/assets/87285532-71b6-4f24-9a92-fdf66a7378b1)
The first two make sense, but the next four are all matches because of their IDs: 20255, 20259, 20251, and 20252.

Reply:
Fixed in latest.